torch/optim/adam.py

```diff
@@ -16,19 +16,24 @@ class Adam(Optimizer):
             running averages of gradient and its square (default: (0.9, 0.999))
         eps (float, optional): term added to the denominator to improve
             numerical stability (default: 1e-8)
-        weight_decay (float, optional): weight decay (L2 penalty) (default: 0)
+        weight_decay (float, optional): weight decay factor (default: 0)
```

Comment:
The appendix of the new version of [the paper](https://arxiv.org/abs/1711.05101) recommends using a "normalized" parametrization of the weight decay that seems to be more stable across datasets and batch sizes.

The problem is that this normalization needs to be aware of the mini batch size, the full batch sizes and the number of epochs when using the cosine annealing of the learning rate.

Maybe this is too much coupling with external information though.

Reply:
@ogrisel I agree that it would be better to go for the simpler version, i.e., without the normalization based on the mini batch size and full batch size as described in the paper.